src/corehost/corehost.cpp

```diff
@@ -2,123 +2,156 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 #include "trace.h"
-#include "pal.h"
 #include "utils.h"
-#include "libhost.h"
+#include "corehost.h"
+#include "fx_ver.h"
+#include "error_codes.h"
+#include "policy_load.h"
+#include <array>
 
-extern int corehost_main(const int argc, const pal::char_t* argv[]);
+#define LIBFXR_NAME MAKE_LIBNAME("hostfxr")
 
-namespace
+bool corehost_t::hostpolicy_exists_in_svc(pal::string_t* resolved_dir)
 {
-enum StatusCode
-{
-    Success                   = 0,
-    CoreHostLibLoadFailure    = 0x41,
-    CoreHostLibMissingFailure = 0x42,
-    CoreHostEntryPointFailure = 0x43,
-    CoreHostCurExeFindFailure = 0x44,
-};
-
-typedef int (*corehost_main_fn) (const int argc, const pal::char_t* argv[]);
-
-// -----------------------------------------------------------------------------
-// Load the corehost library from the path specified
-//
-// Parameters:
-//    lib_dir      - dir path to the corehost library
-//    h_host       - handle to the library which will be kept live
-//    main_fn      - Contains the entrypoint "corehost_main" when returns success.
-//
-// Returns:
-//    Non-zero exit code on failure. "main_fn" contains "corehost_main"
-//    entrypoint on success.
-//
-StatusCode load_host_lib(const pal::string_t& lib_dir, pal::dll_t* h_host, corehost_main_fn* main_fn)
+#ifdef COREHOST_PACKAGE_SERVICING
+    pal::string_t svc_dir;
+    if (!pal::getenv(_X("DOTNET_SERVICING"), &svc_dir))
```

Comment:
We should have a Windows specific check here to get the path to PFx86 to initialize servicing folder and override it with the env var if specified.